scripts/common/common.sh

```diff
@@ -1036,3 +1036,47 @@ function canonical_image_name() {
     fi
     echo "$image"
 }
+
+# wait_for_running_pods waits for pod(s) in a given namspace to transition to Running status
+function wait_for_running_pods() {
+    local namespace=$1
+    local is_job_controller=0
+    local ns_pods=
+    local status=
+    local containers=
+
+    ns_pods=$(kubectl get pods -n "$namespace" -o jsonpath='{.items[*].metadata.name}')
+
+    if [ -z "$ns_pods" ]; then
+        log "No pods found in namespace $namespace."
+        return 0
+    fi
+
+    for pod in $ns_pods; do
+        status=$(kubectl get pod "$pod" -n "$namespace" -o jsonpath='{.status.phase}')
+
+        # determine if pod is manged by a Job
+        if kubectl get pod "$pod" -n "$namespace" -o jsonpath='{.metadata.ownerReferences[*].kind}' | grep -q "Job"; then
+            is_job_controller=1
+        fi
+
+        if [ "$status" != "Running" ] && [ "$status" != "Succeeded" ]; then
+            log "  Pod, $pod, is not ready: $status"
+            return 1
+        fi
+
+        containers=$(kubectl get pod "$pod" -n "$namespace" -o jsonpath="{.spec.containers[*].name}")
+        for container in $containers; do
+            container_status=$(kubectl get pod "$pod" -n "$namespace" -o jsonpath="{.status.containerStatuses[?(@.name==\"$container\")].ready}")
+            
+            # ignore container ready status for pods managed by the Job controller
+            if [ "$container_status" != "true" ] && [ $is_job_controller -eq 0 ]; then
```

Comment:
for equality comparison i always prefer string comparison in bash as you will not get an error for an empty or non numeric value

```suggestion
            if [ "$container_status" != "true" ] && [ "$is_job_controller" = "0" ]; then
```